From f4d2522f60022dca9922193d960ff72c0ed4a7d4 Mon Sep 17 00:00:00 2001
From: Tyler Hoffman <tyler@memfault.com>
Date: Fri, 22 Jul 2022 14:38:56 -0700
Subject: [PATCH] Fix disassemble segfault - check symtab size

---
 opcodes/arm-dis.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opcodes/arm-dis.c b/opcodes/arm-dis.c
index 92cd098d6c..c0789c4e1b 100644
--- a/opcodes/arm-dis.c
+++ b/opcodes/arm-dis.c
@@ -11846,7 +11846,7 @@ mapping_symbol_for_insn (bfd_vma pc, struct disassemble_info *info,
     type = MAP_ARM;
   struct arm_private_data *private_data;
 
-  if (info->private_data == NULL
+  if (info->private_data == NULL  || info->symtab_size == 0
       || bfd_asymbol_flavour (*info->symtab) != bfd_target_elf_flavour)
     return false;
 
-- 
2.36.0

